pymatgen/io/vasp/inputs.py

```diff
@@ -978,7 +976,7 @@ def check_params(self):
                 if type(incar_params[key]).__name__ == "str":
                     # Now we check if this is an appropriate parameter type
                     if incar_params[key] == "float":
-                        if not type(val) not in ["float", "int"]:
+                        if type(val) in ["float", "int"]:
                             warnings.warn(f"{key}: {val} is not real", BadIncarWarning, stacklevel=2)
```

Comment:
This checking mechanism doesn't quite make sense to me... Why a warning is issued when the real type and expected type match?